File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
       if(s=="" || s.length()<1){
        return "";
       }
       int st = 0,e=0;
       for(int i =0;i<s.length();i++){
        int len1 = helper(s,i,i);
        int len2 = helper(s,i,i+1);
        int len = Math.max(len1,len2);
        if(len>e -st){
            st = i-(len -1)/2;
            e = i+len/2;
        }
       }
       return s.substring(st,e+1);
    }
    public int helper(String s , int i ,int j){
        while(i>=0 && j<s.length() && s.charAt(i)==s.charAt(j)){
            i--;
            j++;
        }
        return j-i-1;
    }

}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        maxy = 0
        result = ""
        for i in range(n):
            j = i - 1
            k = i + 1
            while j >= 0 and k < n and s[j] == s[k]:
                if k - j + 1 > maxy:
                    maxy = k - j + 1
                    result = s[j:k+1]
                j -= 1
                k += 1
            j = i
            k = i + 1
            while j >= 0 and k < n and s[j] == s[k]:
                if k - j + 1 > maxy:
                    maxy = k - j + 1
                    result = s[j:k+1]
                j -= 1
                k += 1
        return result or s[0]
        